pkg/controller/controller.go

```diff
@@ -91,20 +114,57 @@ func (c *Controller) ListenAndServe() {
 		return
 	}
 
-	listener, err := tls.Listen(c.Network, c.Address, &tlsConfig)
-	if err != nil {
-		c.startchan <- err
-		return
-	}
-	defer listener.Close()
-	c.startchan <- nil // Controller is listening.
-	logrus.Infof("Listening on %s", c.Address)
-	for {
-		conn, err := listener.Accept()
+	if strings.Contains(c.Address, "https://") {
+		//SSL websocket protocol
+		listener, err := tls.Listen(c.Network, strings.Replace(c.Address, "https://", "", 1), &tlsConfig)
 		if err != nil {
-			logrus.Error(err)
-			continue
+			logrus.Fatal(err)
+		}
+		defer listener.Close()
+		close(c.startchan) // Controller is listening.
+		logrus.Infof("Listening websocket on %s", c.Address)
+
+		s := &http.Server{
+			Handler: myHttpServer{},
+		}
+		for {
+			//start http handler in go routine
+			go func() {
+				err = s.Serve(listener)
+			}()
+			if err != nil {
+				logrus.Error(err)
+			}
+			//manual waiting until handler got connection and global variable wsconn is set by http handler
+			//this not so gracefully but effective ))
+			for {
+				if wsconn != nil {
+					logrus.Infof("Got websocket connection %s", wsconn.RemoteAddr())
```

Comment:
Maybe a little verbose? Debugf instead of Infof?

Reply:
Done in last commit